pool/src/lib.rs

```diff
@@ -688,6 +690,46 @@ impl LendingPoolTrait for LendingPool {
         Ok(())
     }
 
+    fn set_as_collateral(
+        env: Env,
+        who: Address,
+        asset: Address,
+        use_as_collateral: bool,
+    ) -> Result<(), Error> {
+        who.require_auth();
+        let mut user_config = read_user_config(&env, who.clone())?;
+        let reserve_index = read_reserve(&env, asset)?.get_id();
+        if user_config.is_borrowing_any() {
+            if use_as_collateral {
+                assert_with_error!(
+                    &env,
+                    !user_config.is_borrowing(&env, reserve_index),
+                    Error::MustNotHaveDebt
+                );
+            } else if !user_config.is_borrowing(&env, reserve_index) {
```

Comment:
Let's consider rewriting it to not call `set_using_as_collateral` twice. Maybe `match` will help. 

image.png